torch/_prims_common/__init__.py

```diff
@@ -1221,6 +1222,14 @@ def number_type(x: Union[NumberType, torch.SymInt, torch.SymFloat]) -> Type:
         return type(x)
 
 
+def symbol_type(x: sympy.Symbol) -> Type:
+    if x.is_integer:
+        return int
+    else:
+        # NB: Not strictly correct, but we don't support SymPy complex or bool.
+        return float
```

Comment:
Do you know why we don't use `SymInt` and `SymFloat` during lowering?

Reply:
I've answered offline "because Inductor has a SymPy-oriented design". With `specialize_int = True`, these args were plain ints. I've also linked the "Handling int/float inputs/intermediates in Inductor" PDF offline. So I'm assuming it's fine that we handle SymPy Symbols here (as long as the stuff that's stored in `SymInt`s is not needed). cc @ezyang (if you want to expand on this)

Reply:
From the PDF I think the answer is that inductor assumes all sympy expressions evaluate to integers.

Reply:
Well, not exactly, there are some cases where size compute goes through floats and we are OK with it. IDK, it's all very finely balanced.